Shop-1/src/controllers/products.ts

```diff
@@ -0,0 +1,110 @@
+import Product from "../models/Products";
+import Product from "../models/Product";
+import { reqRes, reqResNext } from "../utils/interfaces";
+import { ProductsNotFoundError } from "../utils/errors";
+
+// CREATE
+
+export const postProduct: reqRes = async (req, res) => {
+  try {
+    const product = new Products(req.body);
+    await product.save();
+    res.status(201).send(product);
+  } catch (err) {
+    res.status(400).send("Server Error");
+  }
+};
+
+// READ ALL w/ PARAMS
+
+export const getProductParams: reqResNext = async (req, res, next) => {
+  try {
+    if (Object.keys(req.query).length <= 0) next();
+    else {
+      let Products = await Products.find();
+
+      if (req.query && req.query["max-price"]) {
+        const maxPrice = +req.query["max-price"];
+        Products = Products.filter((Product) => Product.price >= maxPrice);
+      }
+      if (req.query && req.query.includes) {
+        const searchitem = req.query.includes.toString()
+        Products = Product.filter((item) => item.name.includes(searchitem))
+      }
+      if (req.query && req.query.limit) {[
+        const limit = +req.query.limit
+        products = products.slice(0, limit)
+      ]}
+      res.status(200).send(Products)
+    }
+  } catch (err) {
+    res.status(500).send("Server Error");
+  }
+};
+
+// READ ALL
+
+export const getProduct: reqRes = async (req, res) => {
```

Comment:
this shoud be getProducts plural, because we are getting all products. getProduct singular is the next function.